### PR TITLE
processor/k8sattributes: Add resource_attributes to metadata.yaml

### PR DESCRIPTION
**Pull request by mackjmr**
**Description:**

This PR adds the resource attributes added by the k8sattributes processor to the metadata.yaml file in the resource_attributes group. Only the attributes which are added by default (see [documentation](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.79.0/processor/k8sattributesprocessor#configuration)) are marked as enabled.

The default resource attributes defined in metadata.yaml are used to make the default extract::metadata list. 
Resource attributes are exposed through a different configuration interface. This is noted in the metadata.yaml file.

**Link to tracking Issue:** 

#23152

### CODE REVIEW

**Comment by mackjmr** [comment]
I believe the `Skip Changelog` label should be added.

**Review by songy23 — COMMENTED** [review]
Do you also want to mention these?

> Additional container level attributes can be extracted provided that certain resource attributes are provided:
If the container.id resource attribute is provided, the following additional attributes will be available:
k8s.container.name
container.image.name
container.image.tag
If the k8s.container.name resource attribute is provided, the following additional attributes will be available:
container.image.name
container.image.tag
If the k8s.container.restart_count resource attribute is provided, it can be used to associate with a particular container instance. If it's not set, the latest container instance will be used:
container.id (not added by default, has to be specified in metadata)

**Review by songy23 — COMMENTED** [review]
Probably need to wait for https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23136 too.

**Comment by mackjmr** [comment]
Thanks @songy23. I've added the relationship between resource attributes. I will wait to see the outcome of the [issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23136) 👍

**Comment by dmitryax** [comment]
Seems like the attributes are just defined but the generated code is not actually being used. It'll be hard to maintain the default set in two place. Let's use the default resource attributes defined in metadata.yaml to compose the default extract::metadata list?

**Comment by mackjmr** [comment]
@dmitryax Agreed this is a better approach, I've updated so that the default extract::metadata list is pulled from the default resource attributes defined in `metadata.yaml` 👍

File: processor/k8sattributesprocessor/options.go [hunk 1]
```diff
@@ -46,21 +47,81 @@ func withPassthrough() option {
...
+	if defaultConfig.ContainerImageTag.Enabled {
+		attributes = append(attributes, conventions.AttributeContainerImageTag)
+	}
+	if defaultConfig.K8sContainerName.Enabled {
+		attributes = append(attributes, conventions.AttributeK8SContainerName)
+	}
+	if defaultConfig.K8sCronjobName.Enabled {
+		attributes = append(attributes, conventions.AttributeK8SCronJobName)
+	}
+	if defaultConfig.K8sCronjobName.Enabled {
```
Comment (songy23) [inline processor/k8sattributesprocessor/options.go:68]:
```suggestion
	if defaultConfig.K8sDaemonSetName.Enabled {
```

**Review by songy23 — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**